domains/authorization/Proxy.IntegrationTests/V1EndpointTests.cs

```diff
@@ -0,0 +1,225 @@
+        responseContent.Should().Be(orgId);
+    }
+
+    public class V20250101PostTestData : IEnumerable<object[]>
+    {
+        private readonly List<object[]> _data = new List<object[]>
+        {
+            new object[] {"/wallets", new CreateWalletRequest{ PrivateKey = Encoding.ASCII.GetBytes("test") }},
+            new object[] {"/wallets/8229a340-1c9d-46b6-8212-b767e42e02f0/endpoints", new {}},
+            new object[] {"/external-endpoints", new CreateExternalEndpointRequest{TextReference = "Hello", WalletReference = new WalletEndpointReference(){ Endpoint = new Uri("https://test"), Version = 0, PublicKey = "test"}}},
+            new object[] {"/claims", new ClaimRequest{ Quantity = 1, ConsumptionCertificateId = new FederatedStreamId(){ Registry = "test", StreamId = Guid.NewGuid()}, ProductionCertificateId = new FederatedStreamId(){ Registry = "test", StreamId = Guid.NewGuid()}}},
+            // slices is our only insecure endpoint. new object[] {"/slices", new ReceiveRequest{ Quantity = 1, CertificateId = new FederatedStreamId(){ Registry = "test", StreamId = Guid.NewGuid()}, PublicKey = Encoding.ASCII.GetBytes("test"), RandomR = Encoding.ASCII.GetBytes("test"), Position = 1, HashedAttributes = new List<HashedAttribute>()}},
+            new object[] {"/transfers", new TransferRequest{ Quantity = 1, CertificateId = new FederatedStreamId(){ Registry = "test", StreamId = Guid.NewGuid()}, ReceiverId = Guid.NewGuid(), HashedAttributes = new []{"None :D"}}},
+        };
+
+        public IEnumerator<object[]> GetEnumerator() => _data.GetEnumerator();
+
+        IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+    }
+
+    [Theory]
+    [ClassData(typeof(V20250101PostTestData))]
+    public async Task GivenB2C_WhenV20250101EndpointsPostAreUsed_ThenAppendQueryParameterAsWalletOwnerHeader(string endpoint, object requestBody)
+    {
+        // Arrange
+        var orgIds = new List<string> { Guid.NewGuid().ToString() };
+
+        var requestBuilder = Request.Create()
+            .WithPath($"/v1{endpoint}")
+            .WithBody(JsonSerializer.Serialize(requestBody))
+            .UsingPost();
+
+        fixture.WalletWireMockServer
+            .Given(requestBuilder)
+            .RespondWith(
+                Response.Create()
+                    .WithStatusCode(200)
+                    .WithHeader("Content-Type", "text/plain")
+                    .WithBody("{{request.headers.wallet-owner}}")
+                    .WithTransformer()
+            );
+
+        var client = CreateClientWithOrgIds(orgIds);
+
+        // Act
+        client.DefaultRequestHeaders.Add("EO_API_VERSION", "20250101");
+        var response = await client.PostAsync($"{endpoint}?organizationId={orgIds[0]}", new StringContent(JsonSerializer.Serialize(requestBody), Encoding.UTF8, "application/json"));
+        var responseContent = await response.Content.ReadAsStringAsync();
+
+        // Assert
```

Comment:
Nitpick: do we need to have arrange act assert as comments, or could you just space separate between arrange, act, assert?

Reply:
Well. I found more places in code that we blur lines between these. It's nice to have a clear line and easier readability by splitting it up, so we can see where what is being done.